models/articles.model.js

```diff
@@ -0,0 +1,113 @@
+    .then(({ rows }) => {
+        const article = rows[0]
+        if(!article){
+            return Promise.reject({status: 404, msg: `no article found with ID ${articleId}`})
+        }
+        return article //return the article found
+    }   
+)}
+
+const fetchArticles = () => {
+    return db.query(`SELECT 
+            articles.author,
+            articles.title,
+            articles.article_id,
+            articles.topic,
+            articles.created_at,
+            articles.votes,
+            articles.article_img_url,
+            COUNT(comments.comment_id) AS comment_count 
+            FROM articles
+            LEFT JOIN comments
+            ON articles.article_id = comments.article_id
+            GROUP BY articles.article_id
+            ORDER BY articles.created_at DESC;`)
+        .then((result) => {
+        return result.rows
+        })
+        .catch((err) => {
+            next(err)
+        })
+}
+
+const fetchArticleComments = (articleId) => {
+    if(!Number(articleId)) {
+        return Promise.reject({status: 400, msg: "bad request"})
+    }
+    return db.query(`SELECT
+            comment_id,
+            votes,
+            created_at,
+            author,
+            body,
+            article_id
+            FROM comments WHERE article_id = $1
+            ORDER BY created_at DESC;`, [ articleId ])
+        .then((result) => {
+            if(result.rows.length === 0) {
+                return Promise.reject({status: 404, msg: `no comments in article found with ID ${articleId}`})
+            }
+            return result.rows
+        })
+}
+
+const addComment = (articleId, newComment) => {
+    const { username, body } = newComment
+    if(!Number(articleId) || !newComment) {
+        return Promise.reject({status: 400, msg: "bad request"})
+    }
+    if(!newComment.username || !newComment.body) {
+        return Promise.reject({status: 400, msg: "bad request"})
+    }
+    return db.query(`SELECT * FROM users WHERE username = $1;`, [username])    
+        .then(({ rows }) => {
+            if(rows.length === 0) {
+                return Promise.reject({status: 404, msg: "no user exists"})
+        }
+        return db.query(`SELECT * FROM articles WHERE article_id = $1;`, [articleId])    
+        .then(({ rows }) => {
+            if(rows.length === 0) {
+                return Promise.reject({status: 404, msg: `no comments in article found with ID ${articleId}`})
+            }
+        //had to put the below db.query in the above return for it to work
+        return db.query(`INSERT INTO comments (author, body, article_id)
+            VALUES ($1, $2, $3) RETURNING *;`, [username, body, articleId])
+            .then(({ rows }) => {
+                return rows[0]
+            })
+        })
+    })
+}
+
+const newArticleVotes = (articleId, newVotes) => {
+    if(!Number(articleId)) {
+        return Promise.reject({ status: 400, msg: "bad request"})
```

Comment:
These lines are good but pretty DRY - you could have one if statement with or || operators that would allow you to only have to write the Promise.reject line once